mlir/include/mlir/Dialect/Linalg/Transforms/Transforms.h

```diff
@@ -493,6 +493,8 @@ LogicalResult dropUnitDims(RewriterBase &rewriter, GenericOp genericOp,
 struct ElementwiseOpFusionResult {
   Operation *fusedOp;
   llvm::DenseMap<Value, Value> replacements;
+  static llvm::SmallDenseSet<int>
```

Comment:
The `replacements` here is a map from original value to replacement value. You should be able to use that in the caller to get which results are preserved and which are not.

Reply:
Thanks Mahesh for the quick response! 
The `replacements` usage should be after/within applying the fusion.
However, I want to use this function/util as part of the `ControlFusionFn` which is being passed to the 

```
void populateElementwiseOpsFusionPatterns(
    RewritePatternSet &patterns,
    const ControlFusionFn &controlElementwiseOpFusion);
````

Reply:
What kind of decision are you trying to make... do you want the fused output to have the same number of results as the consumer? Then a simple check here would be to check producer has no other uses apart from in the consumer. Are you looking for something more complicated?

Reply:
I think this is one of the examples. However, in the code above this is not the only check.
For example if one of the inits/output operands is being used within the payload Its suited result would also be preserved.
So, changing the logic that you're deciding about which producer's results to preserve, may change my precondition of applying the fusion or not

Reply:
Ok. Maybe a utility would help, but it has to be consistent. Basically the transformation should use the same utility. If there are two places to change, chances are that they break over time. I am not sure how to structure that.

Reply:
ping

Reply:
Actually I am fine with this. I am stamping it cause its just a `static` method of the result object.

Reply:
great, removed it from the result object and declared it as a method in the header.